cost-analyzer/templates/_helpers.tpl

```diff
@@ -157,6 +157,26 @@ will result in failure. Users are asked to select one of the two presently-avail
   {{- end -}}
 {{- end -}}
 
+{{/*
+Cluster Controller API Key check
+*/}}
+{{- define "clusterController.kubecostAPIKeyCheck" -}}
+  {{- if .Values.clusterController.enabled }}
+    {{- if and .Values.clusterController.kubecostAPIKey ((eq .Values.clusterController.createClusterControllerSecret false)) }}
+      {{- fail "\n\nConfiguration error:\nWhen setting clusterController.kubecostAPIKey, clusterController.createClusterControllerSecret must be true." -}}
+    {{- end -}}
+    {{- if and (not .Values.clusterController.createClusterControllerSecret) .Values.clusterController.secretName }}
+      {{- if not (lookup "v1" "Secret" .Release.Namespace .Values.clusterController.secretName) }}
```

Comment:
Nice idea adding some checks in! If we are using the `lookup` function here, do we need to also first check for `.Values.global.platforms.cicd.enabled=true`?

Reply:
fixed, moved to the secret reference in the deployment template.